lambda/update/correction.go

```diff
@@ -4,78 +4,175 @@ import (
 	"github.com/ministryofjustice/opg-data-lpa-store/internal/shared"
 	"github.com/ministryofjustice/opg-data-lpa-store/internal/validate"
 	"github.com/ministryofjustice/opg-data-lpa-store/lambda/update/parse"
+	"strconv"
 	"time"
 )
 
 type Correction struct {
-	DonorFirstNames string
-	DonorLastName   string
-	DonorOtherNames string
-	DonorDob        shared.Date
-	DonorAddress    shared.Address
-	DonorEmail      string
-	LPASignedAt     time.Time
+	Donor       DonorCorrection
+	Attorney    AttorneyCorrection
+	LPASignedAt time.Time
 }
 
+type DonorCorrection struct {
+	FirstNames        string
+	LastName          string
+	OtherNamesKnownBy string
+	DateOfBirth       shared.Date
+	Address           shared.Address
+	Email             string
+}
+
+type AttorneyCorrection struct {
+	Index      *int
+	FirstNames string
+	LastName   string
+	Dob        shared.Date
+	Address    shared.Address
+	Email      string
+	Mobile     string
+	SignedAt   time.Time
+}
+
+const signedAt = "/signedAt"
+
 func (c Correction) Apply(lpa *shared.Lpa) []shared.FieldError {
-	if !c.LPASignedAt.IsZero() && lpa.Channel == shared.ChannelOnline {
-		return []shared.FieldError{{Source: "/signedAt", Detail: "LPA Signed on date cannot be changed for online LPAs"}}
+	if (c.LPASignedAt != time.Time{}) && c.LPASignedAt != lpa.SignedAt && lpa.Channel == shared.ChannelOnline {
```

Comment:
I think using `!x.IsZero()` is cleaner than `x != time.Time{}`

Reply:
also `!t1.Equal(t2)` instead of `t1 != t2` to account for possible timezone mismatches (unless we really trust ourselves to UTC everywhere)